### ISSUE TITLE
Entity and synonym tagging is extremely difficult in Rasa X

### ISSUE THREAD

**Issue by ganeshviyer** [RasaHQ/rasa#5908]
**Rasa version**: 1.7.0

**Rasa SDK version** (if used & relevant): (not used)

**Rasa X version** (if used & relevant): 0.28.1

**Python version**: (not used)

**Operating system** (windows, osx, ...): Mac OS X Catalina 10.15.4; Chrome browser Version 81.0.4044.138 (Official Build) (64-bit)

**Issue**:
I'm having a really hard time tagging entities and synonyms in Rasa X.  Basically, the UI does not represent the state of the system and leaves me very confused as to what I would be pushing to the branch through Integrated Version Control.  

Here are the steps to reproduce -

1. Go to a Rasa X instance and select NLU data under the Training tab.
2. I can see conversations that I know the bot had successfully picked the right entity and synonyms in rasa interactive mode in the command line.  In Rasa X, **they are shown as unannotated to begin with in the list view**.  See below:

![Screenshot 2020-05-27 17 03 59](https://user-images.githubusercontent.com/29989662/83083870-28ba4780-a03c-11ea-8cec-5c2714711e6b.png)

3.  Now if you click on the entity to annotate it, it shows you a modal dialog that tells a different story -

i. There are existing annotations for this example that you see only once you highlight the entity.  The existing annotation should have been seen in the list view.

ii. In the selected/word phrase section, I see two values, despite highlighting one string.  I had highlighted "dogs" but it's showing me "dog" and "dogs".  In fact, the correct entity type and synonym (consistent with rasa interactive mode in the command line) point to the wrong highlight in Rasa X.  

![Screenshot 2020-05-27 17 05 49](https://user-images.githubusercontent.com/29989662/83084179-1ab8f680-a03d-11ea-8e2b-a2306353bfee.png)

iii. When now you actually remove the incorrect entity and synonym pointing to the right highlight (so you should have one left), you go back to the list view still showing no highlights.

iv. But now, when you try to annotate it again, you see the removed entity and synonym values instead of the one you chose to keep.

![Screenshot 2020-05-27 17 13 09](https://user-images.githubusercontent.com/29989662/83084621-57d1b880-a03e-11ea-984e-592c466df119.png)

v. I finally had to modify this by going to the entity dropdown and selecting "petType" as the entity.  This however, closes the modal dialog without letting me add the synonym.

vi. Finally I get to add the synonym by clicking on the highlight and opening the modal dialog again.

4. I finally hit save on the training example and refresh the page and able to see the highlight on the example, followed by the correct annotation on the modal dialog.

This is honestly a lot of effort to correct something because the list to begin with, did not show the right annotations.  And now I'm concerned about pushing changes to my repo because none of the entities highlighted in Rasa X might be right.

Probably not related, but in addition to the above, I'm getting a bunch of repeated console errors and warnings as shown below -

Error: 
```
Error: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Yu (2.27110299.chunk.js:2)
    at va (2.27110299.chunk.js:2)
    at main.f4f835fb.chunk.js:1
    at iu (2.27110299.chunk.js:2)
    at gs (2.27110299.chunk.js:2)
    at t.unstable_runWithPriority (2.27110299.chunk.js:2)
    at Hi (2.27110299.chunk.js:2)
    at vs (2.27110299.chunk.js:2)
    at Qu (2.27110299.chunk.js:2)
    at 2.27110299.chunk.js:2
```

Warning 1:
```
Skipping impossible annotation:
```
Warning 2:
```
Entity (<hidden_entity_name>) spans 151-174, but sentence has length 163
```

I've hidden the entity name in the warning above.  This entity is correctly picked up in rasa interactive mode in the command line.

Warning 3:
```
Editor being replaced.
{previous: "<hidden_message>", new: "<hidden_message"} 
```

I've also hidden the message in the warning above.  The messages are identical and the above error + 3 warnings appear ad infinitum.

If you need me to show this over a video call, I would be happy to.  I really hope there is a swift resolution for this as this is blocking all of our QA efforts.

Many thanks in advance!

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @amn41 will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by abhilasharoy** [comment]
Hey @ganeshviyer, thanks a lot for pointing these out!

From what I understand, entities are not being highlighted accurately on the list view. This probably caused the issues pointed out in point 2, 3 (i) (ii) and (iii). 

Just to confirm, in point 3 (iii), you deleted the row with the entity "dogs" yet the other row got deleted, and the one with "dogs" showed up again in 3 (iv)?


> v. I finally had to modify this by going to the entity dropdown and selecting "petType" as the entity. This however, closes the modal dialog without letting me add the synonym

This should be addressed by this [issue](https://github.com/RasaHQ/rasa-x/issues/2851)


Also tagging @eibhleag and @mjaycub who might understand the error messages better!

**Comment by ganeshviyer** [comment]
Hello @abhilasharoy - thank you for responding!
```
From what I understand, entities are not being highlighted accurately on the list view. 
```
```
Just to confirm, in point 3 (iii), you deleted the row with the entity "dogs" yet the other row got deleted, and the one with "dogs" showed up again in 3 (iv)?
```

Correct on both counts.

**Comment by ganeshviyer** [comment]
@abhilasharoy - any update?

**Comment by abhilasharoy** [comment]
@ganeshviyer This issue is now part of our team's feedback loop, and we'll be looking into it!

@eibhleag and/or @federicotdn will be better able to answer questions about the errors though.

**Comment by mjaycub** [comment]
Thanks @ganeshviyer for this detailed description, it was very helpful for the team as we worked on fixing this issue! 🙏 

I'm closing this issue now as we are shipping improvements and fixes to entity annotation in the `Rasa X 0.40` release that will address your comments!
